external/dockerfile_functions.sh

```diff
@@ -445,7 +446,7 @@ print_maven_install() {
 print_java_tool_options() {
     local file=$1
 
-    echo -e "ENV JAVA_TOOL_OPTIONS=\"-Dfile.encoding=UTF8 -Djava.security.egd=file:/dev/./urandom\"\n" >> ${file}
+    echo -e "ENV JAVA_TOOL_OPTIONS=\"$JAVA_TOOL_OPTIONS -Dfile.encoding=UTF8 -Djava.security.egd=file:/dev/./urandom\"\n" >> ${file}
```

Comment:
JAVA_TOOL_OPTIONS is not defined. No need to do the prefix. Same as other places.

Reply:
Thats not true. JAVA_TOOL_OPTIONS is variable, which is used by JDK, and serves as way of transport any config into JDK.   I - and I doubt that it is jsut me - heavily depends on usage of JAVA_TOOL_OPTIONS, because one declare it once, and have the same setup everywhere. It is very impolite to overwrite it unless there is reason :) I have not found a reason.  If there is, I will drop it of course.

Reply:
This shell script is to  generate Dockerfile. If no JAVA_TOOL_OPTIONS has been explicitly defined or as environment variable  it will just been a empty string. See the Dockerfile generated using this PR https://ci.adoptium.net/view/Test_grinder/job/Grinder/10747/artifact/jvmtest/external/lucene-solr/dockerfile/17/jdk/ubuntu/Dockerfile.hotspot.full

```
ENV JAVA_TOOL_OPTIONS=" -Dfile.encoding=UTF8 -Djava.security.egd=file:/dev/./urandom"
```

Reply:
I see now. Ok. I will move it to different PR, with proepr propagation of JAVA_TOOL_OPTIONS

Reply:
Wait, nope. You confused me. The hunk is doing what it should be doing.  The external JAVA_TOOL_OPTIONS is  expanded in the echo which generates the Dockerfile., and so is correctly beaked in. I will do the separate PR anyway.